### PR TITLE
improve doc for app gw rule settings

### PR DESCRIPTION
**Pull request by nneul**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).    

    --  NN: Documentation update only

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

    --  NN: Documentation update only

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by nneul** [comment]
Current documentation makes it sound like `--disable-rules "123 456 789"` should work, but it's really expecting a list of arguments. Ideally it should support both, but this at least makes docs reflect reality.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 1]
```diff
@@ -525,11 +525,11 @@
           - az application-gateway waf-config list-rule-sets
```
Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: I just noticed that these populator commands are wrong.  They should be `az network application-gateway ...`. They are missing the "network" bit. 😮

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py [hunk 2]
```diff
@@ -525,11 +525,11 @@
...
         - name: --disabled-rule-groups
           short-summary: Space separated list of rule groups to disable. This disables the entire group.
-            To disable specifc rules, use '--disabled-rule-ids'.
+            To disable specifc rules, use '--disabled-rules'.
           populator-commands:
           - az application-gateway waf-config list-rule-sets
         - name: --disabled-rules
-          short-summary: Space separated list of rule IDs to disable.
+          short-summary: List of rule IDs to disable, separate arguments not a single space separated string.
```
Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py]: The original help text is correct and is verified by in-place text.  The proposed text would actually be incorrect. Supplying --disabled-rules twice would result in the second overriding the first. 

"Space separated list of ..." is used consistently throughout the CLI in this context.

**Comment by tjprescott** [comment]
@nneul I just noticed your follow-on comment. 

> Current documentation makes it sound like --disable-rules "123 456 789" should work, but it's really expecting a list of arguments. Ideally it should support both, but this at least makes docs reflect reality

The "Space separated list of ..." phrasing is used everywhere in the CLI to indicate this behavior. I don't see how it suggests that "123 456 789" should work because that isn't a space separated list--that's a single string. --disabled-rules "123" "456" "789" would work (but be entirely unnecessary).

When I read the proposed text, I would expect interpret it as the following:
`--disabled-rules 123 --disabled-rules 456`

This actually *would not* work, as the CLI would simply use the last input.

If you think it is unclear, I would recommend adding as example for the `set` command to demonstrate the proper usage. I have noticed that folks coming from Xplat tend to vastly overquote things, so that might guide them in the right direction.

**Comment by nneul** [comment]
Yeah, I wouldn't expend the 'listed twice' to work... It may be just what someone is used to, but when I see cmd line help or documentation saying that the argument for a given option is a specifically called out as "space separated", I would never interpret that as "list multiple arguments on the command line". 

If it didn't specifically call out "space separated", it would have been more clear, since from the perspective of the actual code - they aren't separated by _anything_, they are coming in as multiple elements in argv[], whereas if it was a single argument, spelling out that it was 'space separated' or 'comma separated' or whatever - is much more necessary/obvious. 

It does appear that the code/usage is consistent throughout though, and once you know that's how az means that statement, it's much more clear. 

(Might have been more intuitive had it returned an actual error instead of just silently failing.)

Curious if that is just me or if anyone else would interpret the same way. Will close this out though. Thank you!

**Comment by tjprescott** [comment]
The change from `--disabled-rule-ids` to `--disabled-rules` is still valid, and the populator commands should say `az network application-gateway ...` instead of `az application-gateway ...` so there's still merit in the PR.  Plus I think adding an example for the `set` command would more accurately clarify what you were trying to say with the help text change.

The reason it calls out "space separated" is because otherwise people might not realize they can put multiple things.

**Comment by nneul** [comment]
Ok, I can do that. If you can go ahead and re-open this and I'll update PR.

**Comment by nneul** [comment]
PR updated.

**Pull request closed without merging**